actions/service.go

```diff
@@ -102,40 +121,199 @@ func (s *Service) loadMatchedActions(ctx context.Context, source Source, spec Ma
+			}
+		}
+		return nil, nil
+	}, db.WithContext(ctx))
+	return err
+}
+
+func (s *Service) UpdateCommitID(ctx context.Context, repositoryID string, runID string, commitID string) error {
+	_, err := s.DB.Transact(func(tx db.Tx) (interface{}, error) {
+		_, err := tx.Exec(`UPDATE actions_runs SET commit_id=$3 WHERE repository_id=$1 AND run_id=$2`,
+			repositoryID, runID, commitID)
+		if err != nil {
+			return nil, fmt.Errorf("update run commit_id: %w", err)
+		}
+		return nil, nil
+	}, db.WithContext(ctx))
+	return err
+}
+
+func (s *Service) GetRunResult(ctx context.Context, repositoryID string, runID string) (*RunResult, error) {
+	res, err := s.DB.Transact(func(tx db.Tx) (interface{}, error) {
+		result := &RunResult{
+			RunID: runID,
+		}
+		err := tx.Get(result, `SELECT event_type, branch_id, source_ref, start_time, end_time, passed, commit_id
+			FROM actions_runs
+			WHERE repository_id=$1 AND run_id=$2`,
+			repositoryID, runID)
+		if err != nil {
+			return nil, fmt.Errorf("get run result: %w", err)
+		}
+		return result, nil
+	}, db.WithContext(ctx), db.ReadOnly())
+	if errors.Is(err, db.ErrNotFound) {
+		return nil, ErrNotFound
+	}
+	if err != nil {
+		return nil, err
+	}
+	return res.(*RunResult), nil
+}
+
+func (s *Service) GetTaskResult(ctx context.Context, repositoryID string, runID string, hookRunID string) (*TaskResult, error) {
+	res, err := s.DB.Transact(func(tx db.Tx) (interface{}, error) {
+		result := &TaskResult{
+			RunID:     runID,
+			HookRunID: hookRunID,
+		}
+		err := tx.Get(result, `SELECT hook_id, action_name, start_time, end_time, passed
+			FROM actions_run_hooks 
+			WHERE repository_id=$1 AND run_id=$2 AND hook_run_id=$3`,
+			repositoryID, runID, hookRunID)
+		if err != nil {
+			return nil, fmt.Errorf("get task result: %w", err)
+		}
+		return result, nil
+	}, db.WithContext(ctx), db.ReadOnly())
+	if errors.Is(err, db.ErrNotFound) {
+		return nil, ErrNotFound
+	}
+	if err != nil {
+		return nil, err
+	}
+	return res.(*TaskResult), nil
+}
+
+func (s *Service) ListRuns(ctx context.Context, repositoryID string, branchID *string, after string) (RunResultIterator, error) {
+	return NewDBRunResultIterator(ctx, s.DB, defaultFetchSize, repositoryID, branchID, after), nil
+}
+
+func (s *Service) ListRunTasks(ctx context.Context, repositoryID string, runID string, after string) (TaskResultIterator, error) {
+	return NewDBTaskResultIterator(ctx, s.DB, defaultFetchSize, repositoryID, runID, after), nil
+}
+
+func (s *Service) PreCommitHook(ctx context.Context, record graveler.HookRecord) error {
+	return s.Run(ctx, record)
+}
+
+func (s *Service) PostCommitHook(ctx context.Context, record graveler.HookRecord) error {
+	// update pre-commit with commit ID if needed
+	err := s.UpdateCommitID(ctx, record.RepositoryID.String(), record.PreRunID, record.CommitID.String())
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
```suggestion
	return s.UpdateCommitID(ctx, record.RepositoryID.String(), record.PreRunID, record.CommitID.String())
```

Reply:
leaving a comment to post merge/commit actions